release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md

```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
+                        	on conflict error("Duplicate Key");
+	io:println(mapB); // error("Duplicate Key")
+}
+```
+
+##### Support for running new strands safely on separate threads
+
+The `isolated` feature has been extended to identify cases where strands created by a `start` action or a named worker can be run safely on separate threads. 
+
+A `call-action-or-expr` is isolated, if the function or method it calls has a type that is isolated and the expression for every argument is an isolated expression. The object whose method is called by the `method-call-expr` or `client-remote-method-call-action` is treated as an argument. When a `start` action is used with a `call-action-or-expr` that is isolated, then the `start` action is allowed in an `isolated` function and the strand created by the `start` action will run in a separate thread from the current thread.
+
+```ballerina
+import ballerina/io;
+
+isolated function f1(int[] arr) returns int[] {
+    return arr.sort();
+}
+
+isolated function f2() {
+    future<int[]> f = start f1([4, 2, 8, 6]); // Now allowed
+}
+
+function f3(int[] arr) returns int[] {
+    return arr.sort();
+}
+
+function f4() {
+    future<int[]> f = start f3([4, 2, 8, 6]);
+}
+
+public function main() {
+    io:println(f3 is isolated function (int[]) returns int[]); // true, f3 is inferred to be isolated
+    io:println(f4 is isolated function ()); // true, f4 is inferred to be isolated
+}
+```
+
+The strand created by a named worker can run on a separate thread from the default worker if the body of the worker satisfies the requirements for an isolated function.
+
+```ballerina
+import ballerina/io;
+
+final int[] & readonly intArr = [4, 2, 8, 6];
+
+isolated function f1(int[] arr) returns int[] {
+    return arr.sort();
+}
+
+isolated function f2() {
+    worker A { // Now allowed
+        int[] f = f1(intArr);
+    }
+}
+
+function f3(int[] arr) returns int[] {
+    return arr.sort();
+}
+
+function f4() {
+    worker A {
+        int[] f = f3(intArr);
+    }
+}
+
+public function main() {
+    io:println(f3 is isolated function (int[]) returns int[]); // true, f3 is inferred to be isolated
+    io:println(f4 is isolated function ()); // true, f4 is inferred to be isolated
+}
+```
+
 ##### Compiler API
 
 - Introduced a set of builders in the Types API, which are used to construct complex types that have varying components
 
 #### Improvements
 
+##### Use `xml` as the contextually-expected type for interpolations in XML template expressions
+
+The contextually-expected type for interpolations in XML template expressions has been changed to `xml`. This ensures that a query expression as an interpolation works as expected. However, it is not an error for the static type to not be a subtype of the contextually-expected type.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    var x1 = xml `<doc>${from int i in 0 ..< 2
+                            select xml `<num>${i}</num>`}</doc>`;
```

Comment:
```suggestion
        select xml `<num>${i}</num>`}</doc>`;
```

Reply:
Isn't it clearer when indented from the start of the query?

Reply:
indentation was a bit off from the next select clause